openml/datasets/dataset.py

```diff
@@ -773,6 +788,31 @@ def get_data(
 
         return data, targets, categorical, attribute_names
 
+    def load_metadata(self, features: bool = False, qualities: bool = False):
+        """Load the missing medata information from the server and store it in the server.
+
+        The purpose of the function is to support lazy loading.
+
+        Parameters
+        ----------
+        features : bool (default=False)
+            If True, load the `self.features` data if not already loaded.
+        qualities: bool (default=False)
+            If True, load the `self.qualities` data if not already loaded.
+        """
+
+        if self.dataset_id is None:
+            raise ValueError(
+                """No dataset id specified. Please set the dataset id.
+                                Otherwise we cannot load metadata."""
+            )
+
+        if features and self.features is None:
+            self.features = _parse_features_xml(_get_features_xml(self.dataset_id))
```

Comment:
Prefer one function call per line, it is much more readable if an (unexpected) error is raised.